director/director.go

```diff
@@ -757,6 +757,7 @@ func registerServeAd(engineCtx context.Context, ctx *gin.Context, sType server_s
 	// Set to ctx for metrics handler downstream
 	ctx.Set("serverName", adV2.Name)
 	ctx.Set("serverWebUrl", adV2.WebURL)
+	ctx.Set("namespacePrefix", adV2.RegistryPrefix)
```

Comment:
Ah, our naming isn't very clear and confuse people. What we want is the federation namespace prefix (i.e. namespace prefix used for object access). Here, the `RegistryPrefix` in `adV2` is the *server* namespace, i.e. namespace for the origin/cache server. What you should look at instead is around line 845 where we iterate over the `adV2.Namespaces`, this is what we want to pass along. You want to pass specifically `NamespaceAdV2.Path` as the values. Also note that an origin can export multiple federation namespace prefixes, so this will be an array. In your following middleware, you want to iterate over the array and add those paths ad different label values, if that makes sense.